test/integration_test.py

```diff
@@ -2054,9 +2054,11 @@ def test_response_security_headers(self):
                         response = requests.get(str(endpoint / path))
                         response.raise_for_status()
                         expected = AzulChaliceApp.security_headers() | expected_headers
-                        # FIXME: Add a CSP header with a nonce value to text/html responses
-                        #        https://github.com/DataBiosphere/azul-private/issues/6
+                        # These paths have a CSP that differs from the default
+                        # by including a randomly generated 'nonce' value, so
+                        # we can't assert the full value of the header.
                         if path in ['/', '/oauth2_redirect']:
+                            self.assertIn('nonce', response.headers['Content-Security-Policy'])
```

Comment:
This is a bit lenient. Let's parse the header value here. Split on `;` then on ` ` and for the entry where the LHS is `nonce` assert that the RHS is a token (length and character set).